dataset/bucket.cpp

```diff
@@ -188,4 +190,37 @@ void append(const VariableView &var0, const VariableConstView &var1) {
     var0.replace_model(combine<Dataset>(var0, var1));
 }
 
+namespace histogram_detail {
+template <class Out, class Coord, class Weight, class Edge>
+using args = std::tuple<span<Out>, span<const Coord>, span<const Weight>,
+                        span<const Edge>>;
+}
+
+Variable histogram(const VariableConstView &data,
+                   const VariableConstView &binEdges) {
+  using namespace scipp::core;
+  using namespace histogram_detail;
+  auto hist_dim = binEdges.dims().inner();
+  const auto &[indices, dim, buffer] = data.constituents<bucket<DataArray>>();
```

Comment:
So is the implementation only for `bucket<DataArray>`?

Reply:
Yes. In principle we could also do this for `bucket<Dataset>`, but I wanted to avoid the effort/code for now.